.github/workflows/trigger-gitlab-pipeline.yml

```diff
@@ -0,0 +1,77 @@
+name: Trigger GitLab pipeline
+on:
+  workflow_call:
+    inputs:
+      triggered-ref:
+        description: 'GitLab project ref to trigger'
+        required: true
+        type: string
+      schedule:
+        description: 'Indication if it is a automatically scheduled request'
+        required: false
+        default: false
+        type: boolean
+      cancel-outdated-pipelines:
+        description: 'If set to true, it will cancel previous pipelines that are running for the same github ref'
+        required: false
+        default: true
+        type: boolean
+    secrets:
+      ci-api-v4-url:
+        description: 'GitLab API v4 root URL'
+        required: true
+      access-token:
+        description: 'GitLab API access token'
+        required: true
+      trigger-token:
+        description: 'GitLab API trigger token'
+        required: true
+      project-id:
+        description: 'GitLab project ID'
+        required: true
+
+jobs:
+  authorize-and-resolve-workflow-ref:
+    environment:  ${{ (github.event_name == 'pull_request_target' &&
```

Comment:
Does this mean each project integrating this must have `External` and `Internal` environments set up in github?

Reply:
Please document those in changelog, readme or commit message

Reply:
will do